src/PointwiseFunctions/AnalyticSolutions/WaveEquation/SemidiscretizedDg.hpp

```diff
@@ -0,0 +1,99 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/ScalarWave/Tags.hpp"  // IWYU pragma: keep
+#include "Options/Options.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/AnalyticSolution.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+class DataVector;
+// IWYU pragma: no_forward_declare Tensor
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace ScalarWave {
+namespace Solutions {
+/*!
+ * \brief An exact solution to the semidiscretized DG ScalarWave
+ * system with an upwind flux
+ *
+ * This solution takes into account the spatial discretization error,
+ * and so should show convergence in time integration accuracy to
+ * roundoff at any resolution.
+ *
+ * \warning This is not really a pointwise function, as the solution
+ * depends on the spatial discretization.  It will only work on a
+ * periodic domain of length \f$2 \pi\f$ (or an integer multiple) with
+ * equally sized linear elements.
+ */
+class SemidiscretizedDg : public MarkAsAnalyticSolution {
+ public:
+  using tags = tmpl::list<ScalarWave::Pi, ScalarWave::Phi<1>, ScalarWave::Psi>;
+
+  struct Harmonic {
+    using type = int;
```

Comment:
can this be a size_t?

Reply:
I guess, but why?

Reply:
My motivation is whether there are any constraints on the allowable values of Harmonic.  When I made the comment I thought it could be any non-negative integer in which case our standard practice has been to make them `size_t`.  But looking at the code, maybe there are no constraints

Reply:
Yes, any integer is valid.  (In fact, it turns out that aliasing causes the solution to be periodic in this parameter.)